main.tf

```diff
@@ -15,11 +15,24 @@
 */
 
 locals {
-  eni_configs = [for e in var.pods_subnets : {
-    name           = e.availability_zone
-    subnet         = e.id
-    securityGroups = [aws_eks_cluster.quortex.vpc_config[0].cluster_security_group_id]
-  }]
+  vpc_cni_configuration_values = var.custom_networking ? jsonencode(
+    {
+      "env" : {
+        "AWS_VPC_K8S_CNI_CUSTOM_NETWORK_CFG" : "true",
+        "ENI_CONFIG_LABEL_DEF" : "topology.kubernetes.io/zone"
+      }
```

Comment:
Nice PR thanks !
The issue I have with this is that before we could override these without having to compute manually eniConfig.
Since vpc-cni add-on is typed, maybe we could add a typed version which would be an object with defaults ?